oci8_sql_test.go

```diff
@@ -1475,6 +1475,104 @@ func benchmarkPrefetchSelect(b *testing.B, prefetchRows int64, prefetchMemory in
 	}
 }
 
+// TestSelectParallelWithStatementCaching checks parallel select from dual but with statement caching enabled
+func TestSelectParallelWithStatementCaching(t *testing.T) {
+	if TestDisableDatabase {
+		t.SkipNow()
+	}
+	db := testGetDB("?stmt_cache_size=100")
+	if db == nil {
+		t.Fatal("db is null")
+	}
+
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(50)
+
+	for i := 0; i < 50; i++ {
+		go func(num int) {
+			defer waitGroup.Done()
+
+			selectNumFromDual(t, db, num)
+		}(i)
+	}
+
+	waitGroup.Wait()
+}
+
+// selectNumFromDual will execute a "select :1 from dual" where the parameter is the num param of this function
+func selectNumFromDual(t testing.TB, db *sql.DB, num int) {
```

Comment:
Would change num to a float so this function could be used in cases for int or float. The caller of the function can decide if it is alright to cast int to float or not.